lib/middleware/routes-static/routes-static.js

```diff
@@ -1,28 +1,60 @@
 const path = require('path')
 const express = require('express')
+const { stat } = require('fs').promises
 
 const staticOptions = {
   index: false,
   immutable: true,
   maxAge: '28d'
 }
 
+async function locateAssetFolders(componentDirs) {
+  const possibleAssetFolders = componentDirs.map(({sourcePath}) => {
+    return path.resolve(sourcePath, 'assets')
+  })
+
+  const res = possibleAssetFolders.filter(async (folder) => {
+    let isDirectory;
+
+    try {
+      isDirectory = (await stat(folder)).isDirectory()
+    } catch (err) {
+      // Don't serve asset folders which do not exist
+    }
+
+    return isDirectory;
+  })
+
+  return Promise.all(res)
+}
+
 const router = express.Router()
 const serveStatic = (assetsUrlPrefix, staticPath) => {
   router.use(assetsUrlPrefix, express.static(staticPath, staticOptions))
 }
 
-const routesStatic = (assetsUrlPrefix, staticPaths = [], serviceDir) => {
+async function routesStatic(options = {}) {
+  const {
+    assetsUrlPrefix,
+    staticPaths = [],
+    serviceDir,
+    componentDirs = []
+  } = options
+
   staticPaths.forEach(staticPath => {
     // TODO: add ability to pass an explicit url prefix { urlPrefix: '/foo', dir}
     serveStatic(assetsUrlPrefix, staticPath)
   })
 
-  let nodeModules = path.join(serviceDir, 'node_modules')
+  if (componentDirs.length) {
+    for (let directory of await locateAssetFolders(componentDirs)) {
+      serveStatic(assetsUrlPrefix, directory)
+    }
+  }
 
-  serveStatic(assetsUrlPrefix, path.resolve(nodeModules, '@ministryofjustice', 'fb-components-core', 'assets'))
-  serveStatic(assetsUrlPrefix, path.resolve(nodeModules, 'govuk-frontend', 'assets'))
-  serveStatic(assetsUrlPrefix, path.resolve(nodeModules, 'govuk-frontend'))
+  // TODO: Why are we serving this folder?
+  let nodeModulesPath = path.join(serviceDir, 'node_modules')
+  serveStatic(assetsUrlPrefix, path.resolve(nodeModulesPath, 'govuk-frontend'))
```

Comment:
@solidgoldpig not sure of this one?

Reply:
I'm pretty sure this a hangover from the very first iteration. We're not using either `all.js` or `common.js` files, so this should be zapped.